Fix/issues 5

alex-melnyk opened the pull request.

alex-melnyk committed:
Modifications applied.

alex-melnyk committed:
Fresh fixes.

IgorGanapolsky commented:
What is the description of this pull request, and how does it relate to "Modifications Applied" pull request?

IgorGanapolsky commented:
Unable to launch app.  Crash:
> E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-1
    Process: com.igorganapolsky.vibratingwatchapp, PID: 3241
    java.lang.IllegalStateException: Room cannot verify the data integrity. Looks like you've changed schema but forgot to update the version number. You can simply fix this by increasing the version number.
        at android.arch.persistence.room.RoomOpenHelper.checkIdentity(RoomOpenHelper.java:135)
        at android.arch.persistence.room.RoomOpenHelper.onOpen(RoomOpenHelper.java:115)
        at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onOpen(FrameworkSQLiteOpenHelper.java:151)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:310)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:194)
        at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:96)
        at android.arch.persistence.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:54)
        at android.arch.persistence.room.RoomDatabase.compileStatement(RoomDatabase.java:244)
        at android.arch.persistence.room.SharedSQLiteStatement.createNewStatement(SharedSQLiteStatement.java:65)
        at android.arch.persistence.room.SharedSQLiteStatement.getStmt(SharedSQLiteStatement.java:72)
        at android.arch.persistence.room.SharedSQLiteStatement.acquire(SharedSQLiteStatement.java:87)
        at com.igorganapolsky.vibratingwatchapp.domain.local.TimersDao_Impl.disableAll(TimersDao_Impl.java:155)
        at com.igorganapolsky.vibratingwatchapp.domain.WatchRepository.lambda$disableAll$5(WatchRepository.java:67)
        at com.igorganapolsky.vibratingwatchapp.domain.-$$Lambda$WatchRepository$UOke5DJ4yvgXckQg-vyJI_madU8.run(Unknown Source:2)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)

IgorGanapolsky reviewed (changes requested):
.

alex-melnyk committed:
Fresh fixes.

alex-melnyk committed:
New fixes.

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/VibratingWatchApp.java:
```diff
@@ -0,0 +1,46 @@
+import com.igorganapolsky.vibratingwatchapp.manager.vibration.WatchBeepManager;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
```
unused import

IgorGanapolsky commented:
Explain difference between **core** package and **domain** package

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/core/util/ViewModelFactory.java:
```diff
@@ -0,0 +1,50 @@
+    private static ViewModelFactory factory;
+
+    private Repository repository;
+    private CountdownManager countdownManager;
```
can be `final` vars

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/WatchRepository.java:
```diff
@@ -0,0 +1,69 @@
+public class WatchRepository implements Repository {
+
+    private TimersDatabase timerDb;
+    private ExecutorService executor;
```
can be `final` vars

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/local/TimersDao.java:
```diff
@@ -0,0 +1,38 @@
+    LiveData<List<TimerEntity>> getAll();
+
+    @Query("SELECT * FROM timers WHERE id = :id")
+    LiveData<TimerEntity> observeById(int id);
```
Where is this method used?

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/BuzzSetup.java:
```diff
@@ -0,0 +1,28 @@
+
+    private Type buzzType;
+    private int buzzCount;
+    private int buzzTime;
```
can be `final` vars

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/CountData.java:
```diff
@@ -0,0 +1,42 @@
+        this.currentTime = currentTime;
+    }
+
+    public void setAnimationNeeded(boolean animationNeeded) {
```
where is this method used?

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/domain/model/TimerModel.java:
```diff
@@ -0,0 +1,161 @@
+        this(0, 0, 0);
+    }
+
+    public TimerModel(int hours, int minutes, int seconds) {
```
can be `private`

IgorGanapolsky commented on app/src/main/java/com/igorganapolsky/vibratingwatchapp/manager/timer/CountdownManager.java:
```diff
@@ -0,0 +1,36 @@
+
+    int getActiveId();
+
+    boolean isHasMoreRepeats();
```
what is the purpose of this method?

IgorGanapolsky commented:
Would it make sense to put **manager** and **vibration** packages in `domain` or `core` packages?  Or perhaps make them into separate gradle modules.  I'm trying to understand the architecture followed here...

IgorGanapolsky commented:
Thank you for externalizing values into **strings.xml** and **dimens.xml** 👍

IgorGanapolsky commented:
Let's improve the speed of launching these screens:
https://drive.google.com/open?id=1hAE9OoLJpF2tK_rFMofIkjJv2Y1hT02J

IgorGanapolsky commented:
Unable to tap on the RecyclerView elements:
https://drive.google.com/open?id=1-68aY5WvOb2nuAuC90I3MvZFGBDzMg78

IgorGanapolsky reviewed (changes requested):
Please address PR remarks

IgorGanapolsky commented:
Looks like this **circle dot** doesn't accurately reflect the number:
![device-2019-03-15-170132](https://user-images.githubusercontent.com/201209/54461830-26741200-4744-11e9-8358-0c6bc9732e0c.png)


Also, can we make this **circle dot** bigger?  It is difficult for the user to drag it now.

IgorGanapolsky commented:
Request: is it possible to make swiping **left-to-right** on this screen easier?  Right now, the user cannot exit this screen unless they carefully swipe in the proper area:
https://drive.google.com/open?id=1khwpmK7XtqBKefkwXsKkMk_nWj-OW9WV

IgorGanapolsky commented:
Pressing the stop button a running timer should reset it to the correct value, not **### 00:00:00**:
https://drive.google.com/open?id=1NuIJH9BXJ0sgW6M-8Xl8jSjda_LZMveg

IgorGanapolsky commented:
Looks like the **reset** button completely doesn't work:
https://drive.google.com/open?id=1PdfXPEzejdQgkWnUN5b61XIqTU6E7F8s

IgorGanapolsky commented:
Shouldn't there be indicators on this screen to tell the user which timer is running and which is not?
![device-2019-03-15-173652](https://user-images.githubusercontent.com/201209/54463326-1579cf80-4749-11e9-84ba-a49025c721fc.png)

IgorGanapolsky commented:
Another issue: when the timer buzzes, user is currently not brought to the screen to _see it_ or _stop it_.  Instead, the watch screen remains off, or on whatever other app is open at that time, and the watch is buzzing.

**Expected behavior:** just like any good timer app, a triggered timer should open the triggered timer screen for the user
**Actual behavior:** nothing happens other than the watch is buzzing

IgorGanapolsky commented:
There is also a warning in Logcat:
> W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.igorganapolsky.vibratingwatchapp flg=0x4000010 (has extras) } to com.android.vending/com.google.android.finsky.packagemanager.impl.PackageMonitorReceiverImpl$RegisteredReceiver

IgorGanapolsky commented:
Incorrect flow in the app.  Steps to reproduce:
1) Create two timers
2) Launch & start first timer from Timers List screen
3) Go back to Timers List screen
4) Open the second timer, edit its settings

**Expected result:**
you should be back on Timers List screen
**Actual result:**
You are on the first timer's activation/status screen.  After that I cannot even open the **second** timer at all (can only open **first** timer)!
https://drive.google.com/open?id=1hdqVZuqOFQukFNDb1yRWI5z7GHGPN6c9

IgorGanapolsky commented:
The **stop** button doesn't work:
https://drive.google.com/open?id=1YsZp95Fu3Mt9BXjWpz4fnU1xX59ItAPG

IgorGanapolsky commented:
Warnings in Logcat like this should be fixed:
>  W/ActivityManager: Duplicate finish request for ActivityRecord{4d90de9 u0 com.igorganapolsky.vibratingwatchapp/.presentation.main.TimerListActivity

IgorGanapolsky commented:
This warning should be fixed:
>  W/ActivityManager: Activity pause timeout for ActivityRecord{d9d18a1 u0 com.igorganapolsky.vibratingwatchapp/.presentation.settings.SetTimerActivity

IgorGanapolsky commented:
Why is Logcat showing this?
> I/Choreographer: Skipped 46 frames!  The application may be doing too much work on its main thread.

What is going on the main thread?

IgorGanapolsky commented:
Another issue:
after editing a timer's config, the app goes back to config screen (instead of Timers List screen)

IgorGanapolsky commented:
What the heck is going on here???
https://drive.google.com/open?id=1QXG_C2660u-9pK0GR9O3NG6LKwrxgtZw

yuliiatereshchenko commented:
> Looks like this **circle dot** doesn't accurately reflect the number:
> ![device-2019-03-15-170132](https://user-images.githubusercontent.com/201209/54461830-26741200-4744-11e9-8358-0c6bc9732e0c.png)
> 
> Also, can we make this **circle dot** bigger? It is difficult for the user to drag it now.

1 the circle dot reflect the current number for H. the current H number is less than 1, that is why it reflect 0
2 yes,  but its an improvement

yuliiatereshchenko commented:
> Looks like the **reset** button completely doesn't work:
> https://drive.google.com/open?id=1PdfXPEzejdQgkWnUN5b61XIqTU6E7F8s

1 please, could you specify which journey describes behaviour of this button?

yuliiatereshchenko commented:
> Another issue: when the timer buzzes, user is currently not brought to the screen to _see it_ or _stop it_. Instead, the watch screen remains off, or on whatever other app is open at that time, and the watch is buzzing.
> 
> **Expected behavior:** just like any good timer app, a triggered timer should open the triggered timer screen for the user
> **Actual behavior:** nothing happens other than the watch is buzzing

1 please, could you specify which journey describes that it should be brought to the screen to _see it_ or _stop it_?

yuliiatereshchenko commented:
> after editing a timer's config, the app goes back to config screen (instead of Timers List screen)

1 there is no information in Journey 13 where should be user redirected after configuration is done

Commit:
bump tools gradle

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
Migrate to AndroidX

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
androidx

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
Lint & static analysis cleanup

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
Code cleanup and PR comments

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
Kotlin refactoring

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
Some changes

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

IgorGanapolsky commented on app/build.gradle:
```diff
@@ -32,30 +30,39 @@ android {
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+    implementation 'androidx.recyclerview:recyclerview:1.0.0'
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'com.google.android.material:material:1.0.0'
```
```what is this```

Commit:
Fixup issues

Upgrade SeekArc library dependency.
Use Lint static analysis.
Update Gradle scripts.
Introduce Kotlin

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
Reorganize code, add backup

Signed-off-by: Igor Ganapolsky <igor.ganapolsky@hulu.com>

Commit:
Dep on room-coroutines

IgorGanapolsky committed:
More fixes, reorganization, conversion to Kotlin

Improve the codebase to make it more testable and professional.

IgorGanapolsky committed:
Merge branch 'fix/issues_5' of github.com:IgorGanapolsky/android-watch-app into fix/issues_5

# Conflicts:
#	app/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/VibratingWatchApp.java
#	buildSrc/src/main/kotlin/com/igorganapolsky/vibratingwatchapp/buildsrc/Dependencies.kt

IgorGanapolsky committed:
Fix build scripts

IgorGanapolsky committed:
Upgrade build and fix more issues

Rename view names in layout files.

IgorGanapolsky committed:
Fix some compilcation issues and build

IgorGanapolsky committed:
Introduce nav graph

IgorGanapolsky committed:
Fix up and rename classes & properties.

IgorGanapolsky committed:
Make first screen launch successfully!

IgorGanapolsky committed:
Modify styles
